server/controllers/project.controller.js

```diff
@@ -63,12 +64,24 @@ export function getProject(req, res) {
     });
 }
 
-export function deleteProject(req, res) {
-  Project.remove({ _id: req.params.project_id }, (err) => {
-    if (err) {
-      return res.status(404).send({ message: 'Project with that id does not exist' });
+function deleteFilesFromS3(files) {
+  files.forEach(file => {
+    if (!(file.url === undefined)) {
+      deleteObjectFromS3(file.url);
     }
-    return res.json({ success: true });
+  });
+}
+
+export function deleteProject(req, res) {
+  // Deleting project files form S3
+  Project.findById(req.params.project_id, (err, project) => {
```

Comment:
I left a thorough answer for this in another comment, but the short answer is callbacks 😄